Changes to Parser unit tests

RC-Lee opened the pull request:
- Moved old backend test setup to Parser
- Uncommented url-parser unit tests
- Modified wiki-feed-test fetch to use mocked Satellite fetch

## Issue This PR Addresses
- `pnpm test src/api/parser` passes all unit tests
- I've uncommented out the code in `url-parser.test` they pass fine, so maybe we could keep them?
- updated the test setup to include the old backend setup so we can update the Parser test configs
- I've changed the `fetch` in `wiki-feed-parser.test` to use the `{ fetch }` from Satellite (which is mocked from the test setup). I figured this seemed more appropriate since we're using the `fetch` from Satellite in the actual file now

TueeNguyen commented on src/api/parser/test/url-parser.test.js:
```diff
@@ -21,42 +21,42 @@ describe('url-parser tests', () => {
-  //   expect(result).toBe(expectedReturn2);
-  // });
+  test('urlParser with double ports', () => {
+    const result = urlParser(testUrlWithPort, testport);
+    expect(result).toBe(expectedReturn1);
+  });
+
+  test('urlParser with 1 string port', () => {
+    const result = urlParser(testUrlWithoutPort, testport);
+    expect(result).toBe(expectedReturn1);
+  });
+
+  test('urlParser with invalid url', () => {
+    const result = urlParser('', testport);
+    expect(result).toBe(null);
+  });
+
+  test('urlParser with url with no port and no port', () => {
+    const result = urlParser(testUrlWithoutPort);
+    expect(result).toBe(expectedReturn2);
+  });
+
+  test('urlPaser with port being a null value', () => {
+    const result = urlParser(testUrlWithoutPort, null);
+    expect(result).toBe(expectedReturn2);
+  });
+
+  test('urlPaser with port being an integer', () => {
+    const result = urlParser(testUrlWithoutPort, Number(testport));
+    expect(result).toBe(expectedReturn1);
+  });
+
+  test('urlPaser with port being an invalid string', () => {
+    const result = urlParser(testUrlWithoutPort, 'invalid');
+    expect(result).toBe(expectedReturn2);
+  });
+  test('urlPaser with an out of range port', () => {
+    const result = urlParser(testUrlWithoutPort, 999999);
+    expect(result).toBe(expectedReturn2);
+  });
```
These ones I've changed it [test.concurrent.each`table`](https://jestjs.io/docs/api#2-testconcurrenteachtablename-fn-timeout) to be more intuitive

RC-Lee replied:
Makes sense, lemme take them out

RC-Lee committed:
Changes to Parser unit tests
- Moved backend test setup to Parser test setup
- Cleaned up url-parser tests
- Refactored wiki-feed-parser tests to use mocked fetch from Satellite
